### PR TITLE
Generate nested navigation when scraping GitBook and ReadMe

### PR DESCRIPTION
**Pull request by tobias-mintlify**
# Summary

Navigation was flattened before in the `mint.json` the CLI generates. This correctly generates nested menus when they are being used.

# Test Plan

Run `mintlify scrape-section https://docs.readme.com/` and `mintlify scrape-section https://docs.gitbook.com/` in empty folders.

### CODE REVIEW

File: packages/mintlify/src/scraping/getSitemapLinks.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import axios from "axios";
 
+// Not in use.
+// Gets all links in a sitemap.
```
Comment (hanywang2) [inline packages/mintlify/src/scraping/getSitemapLinks.ts:4]: If it's not in use, why is it not deleted?

**Review by hanywang2 — COMMENTED** [review]
One comment. Other than that it lgtm

**Review by hanywang2 — CHANGES_REQUESTED** [review]
Build failiures

**Review by hahnbeelee — APPROVED** [review]
lgtm!

File: client/src/config.ts [hunk 2]
```diff
@@ -10,11 +10,11 @@ import configJSON from './config.json';
 
 export const config: Config = configJSON;
 
-export type Page = string | Navigation;
```
Comment (hahnbeelee) [inline client/src/config.ts:13]: import Page

File: packages/mintlify/src/scraping/getSitemapLinks.ts (hunk 1 above)
Comment (tobias-mintlify) [reply packages/mintlify/src/scraping/getSitemapLinks.ts:4]: It might be useful / needed in the near future. We can delete it in a few months if not.

**Pull request merged**